bangazonreports/views/orders/completedorder.py

```diff
@@ -0,0 +1,64 @@
+"""Module for generating games by user report"""
+import sqlite3
+from django.shortcuts import render
+from bangazonapi.models import Order
+from bangazonreports.views import Connection
+
+
+def completed_order_list(request):
+    """Function to build an HTML report of games by user"""
+    if request.method == 'GET':
+        # Connect to project database
+        with sqlite3.connect(Connection.db_path) as conn:
+            conn.row_factory = sqlite3.Row
+            db_cursor = conn.cursor()
+
+            # Query for all games, with related user info.
+            db_cursor.execute("""
+                SELECT
+                    o.id as order_id,
+                    pt.merchant_name,
+                    u.first_name || ' ' || u.last_name AS full_name,
+                    SUM(p.price) as total
+                FROM
+                    bangazonapi_order o
+                JOIN
+                    bangazonapi_customer c ON c.id = o.customer_id
+                JOIN
+                    auth_user u ON c.user_id = u.id
+                JOIN
+                    bangazonapi_payment pt ON pt.id = o.payment_type_id
+                JOIN
+                    bangazonapi_orderproduct op ON op.order_id = o.id
+                JOIN
+                    bangazonapi_product p ON p.id = op.product_id
```

Comment:
While this looks like it gives you back the right data, it needs a where clause to check that the payment_type_id is not null

Reply:
yeah, the payment JOIN seemed to handle all the null cases for me, so I didn't think it was needed.